### PR TITLE
feature: Automatically pass SOURCE_DATE_EPOCH to Linux containers

### PR DESCRIPTION
**Pull request by dalcinl**
Support for reproducible builds: https://reproducible-builds.org/docs/source-date-epoch/

### CODE REVIEW

**Comment by dalcinl** [comment]
@joerick I listed you as a co-author of the commit, as the actual implementation follows your suggestion in #1584. Feel free to amend the commit with your digital signature and force push. Otherwise, I can remove the Co-authored-by line if you prefer.

**Comment by joerick** [comment]
Thank you @dalcinl! A test for this somewhere would be good - probably an integration test would be appropriate to ensure we never lose this. Let me know if you need any help/pointers!

**Comment by dalcinl** [comment]
@joerick Done, but I got fed up fighting pre-commit and its code uglification. Maybe I should do it the following way, but I'll not keep wasting my time without your final word on it.

```python
env_manager = modified_os_environ(CIBUILDWHEEL="1", SOURCE_DATE_EPOCH="42")
oci_manager = OCIContainer(engine=container_engine, image=DEFAULT_IMAGE)
with env_manager, oci_manager:
   ...
```

PS: As you did not have a test for `CIBUILDWHEEL` being passed to containers, I took the liberty of adding a test for it.

File: unit_test/oci_container_test.py
Comment (joerick) [inline unit_test/oci_container_test.py:83]: There's a feature in pytest for this - [monkeypatch.setenv](https://docs.pytest.org/en/7.1.x/how-to/monkeypatch.html). It sets the env var for the duration of the test.

**Review by joerick — COMMENTED** [review]
Thanks!

> PS: As you did not have a test for CIBUILDWHEEL being passed to containers, I took the liberty of adding a test for it.

There is an integration test for this in `/test`, but good to have it here too.

File: unit_test/oci_container_test.py
Comment (dalcinl) [reply unit_test/oci_container_test.py]: Thanks for the tip, I'm not a pytest user.

**Comment by dalcinl** [comment]
> There is an integration test for this in `/test`, but good to have it here too.

As I'm not a pytest user, it was not clear to me if these tests were actually exercising the passing through the docker/podman wall.

At this point, I believe I've addressed all your comments. Please note that I made a relatively small test change since your last review: rather than modifying an existing test function, now I'm adding a new one to exercise environment passing.

**Comment by dalcinl** [comment]
@joerick I believe this PR is ready for merging. There are CI workflows waiting for approval. The Cirrus CI failure is most likely unrelated and may need a manual restart.

**Review by joerick — APPROVED** [review]
Looks good! Thank you

**Pull request merged**
